### PR TITLE
Fm/features/add comments (check after images PR!!!)

### PR DESCRIPTION
**Pull request by frankmangone**

### CODE REVIEW

**Comment by gastonmorixe** [comment]
@frankmangone changeed the base of this PR to the images one

File: src/entities/calendarEvent/calendarEvent.service.ts [hunk 1]
```diff
@@ -76,8 +76,8 @@ export class CalendarEventService extends BaseService<CalendarEvent> {
   buildListQuery(filters: ListCalendarEventsInput): TListQuery {
     const conditions: TListCondition[] = []
```
Comment (gastonmorixe) [inline src/entities/calendarEvent/calendarEvent.service.ts:77]: I'd name it just `TConditions`

File: src/entities/comment/comment.model.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { Schema, Prop } from '@nestjs/mongoose' // { Prop, Schema, SchemaFactory }
+import { ObjectType, Field } from '@nestjs/graphql'
+import mongoose from 'mongoose'
+import * as Utils from '../../utils/Model'
+
+import { User } from '../user/user.model'
```
Comment (gastonmorixe) [inline src/entities/comment/comment.model.ts:6]: add an empty line

File: src/entities/comment/comment.model.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Schema, Prop } from '@nestjs/mongoose' // { Prop, Schema, SchemaFactory }
+import { ObjectType, Field } from '@nestjs/graphql'
+import mongoose from 'mongoose'
+import * as Utils from '../../utils/Model'
```
Comment (gastonmorixe) [inline src/entities/comment/comment.model.ts:4]: group this line with line 6

File: src/entities/comment/comment.model.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import { User } from '../user/user.model'
+@ObjectType()
+@Schema({
+  timestamps: true,
+})
+export class Comment extends Utils.BaseModel {
+  @Field(() => String)
+  @Prop(() => String)
+  content: string
```
Comment (gastonmorixe) [inline src/entities/comment/comment.model.ts:14]: add a validation of min length 2 chars using `class-validator` deco

File: src/entities/discussion/discussion.inputs.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import * as GQL from '@nestjs/graphql' // { Field, ObjectType, ID }
+import mongoose from 'mongoose'
+
+import { VirtualLocationInput } from '../virtualLocation/virtualLocation.inputs'
+import { CommentInput } from '../comment/comment.inputs'
+@GQL.InputType()
```
Comment (gastonmorixe) [inline src/entities/discussion/discussion.inputs.ts:6]: add empty line

File: src/entities/discussion/discussion.model.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+export class Discussion extends Utils.BaseModel {
+  constructor(subject: Subject) {
+    super()
+    this.subject = subject
+  }
+
+  @GQL.Field(() => GQL.ID)
+  _id: mongoose.Types.ObjectId
+
+  @GQL.Field(() => String, { nullable: true })
```
Comment (gastonmorixe) [inline src/entities/discussion/discussion.model.ts:33]: why is this nullable?

File: src/entities/discussion/discussion.model.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  @DB.Prop(() => String)
+  title: string
+
+  @GQL.Field(() => String, { nullable: true })
```
Comment (gastonmorixe) [inline src/entities/discussion/discussion.model.ts:37]: why is it nullable? may you make it false and add a validation of at lease 2 chars?

File: src/entities/discussion/discussion.model.ts (hunk 6 above)
Comment (gastonmorixe) [reply src/entities/discussion/discussion.model.ts:33]: may you make it false and add a validation of at lease 2 chars?

File: src/entities/discussion/discussion.model.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  /*
+   * References for subdocuments in decorators:
+   * https://stackoverflow.com/questions/62704600/mongoose-subdocuments-in-nest-js
+   * https://www.javaer101.com/en/article/40315274.html
+   */
+  @GQL.Field(() => [Comment], { nullable: true })
+  @DB.Prop({
+    type: [CommentSchema],
+    required: false,
+    default: [],
```
Comment (gastonmorixe) [inline src/entities/discussion/discussion.model.ts:50]: why setting a default?

File: src/entities/discussion/discussion.module.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+      {
+        name: Subject.name,
+        schema: SubjectSchema,
+      },
+    ]),
+  ],
+  providers: [
+    DiscussionService,
+    DiscussionResolver,
+    // {
```
Comment (gastonmorixe) [inline src/entities/discussion/discussion.module.ts:26]: you can remove this

**Review by gastonmorixe — CHANGES_REQUESTED** [review]
Awesome u r killing it @frankmangone  :) requested some changes then LGTM

File: src/entities/discussion/discussion.model.ts (hunk 8 above)
Comment (frankmangone) [reply src/entities/discussion/discussion.model.ts:50]: why not?

**Pull request merged**